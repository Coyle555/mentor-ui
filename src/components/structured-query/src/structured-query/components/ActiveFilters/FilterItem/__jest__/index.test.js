jest.mock('react-intl', () => {
	return {
		FormattedDate: (props) => <div>Date: {JSON.stringify(props)}</div>,
		FormattedTime: (props) => <div>Time: {JSON.stringify(props)}</div>
	};
});

import React from 'react';
import { FilterItem } from '../index';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';

afterEach(cleanup);

test('Rendering default filter item', () => {
	const tree = renderer.create(<FilterItem />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Filter item with child object', () => {
	const tree = renderer.create(
		<FilterItem>{{ label: 'Foo', operator: 'Bar', value: 'Baz' }}</FilterItem>
	).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Filter item of type datetime', () => {
	const tree = renderer.create(
		<FilterItem type="datetime">
			{{ label: 'Foo', operator: 'Bar', value: '1995-10-22' }}
		</FilterItem>
	).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Filter item of type date', () => {
	const tree = renderer.create(
		<FilterItem type="date">
			{{ label: 'Foo', operator: 'Bar', value: '1995-10-22' }}
		</FilterItem>
	).toJSON();

	expect(tree).toMatchSnapshot();
});

test('On remove click callback', () => {
	const tbody = document.createElement('tbody');
	const onRemove = jest.fn();

	const { getByText } = render(
		<FilterItem onRemove={onRemove}>
			{{ label: 'Foo', operator: 'Bar', value: 'Baz' }}
		</FilterItem>,
		{ container: document.body.appendChild(tbody) }
	);

	fireEvent.click(getByText('Clear'));
	expect(onRemove).toHaveBeenCalledWith({ label: 'Foo', operator: 'Bar', value: 'Baz' });
});
