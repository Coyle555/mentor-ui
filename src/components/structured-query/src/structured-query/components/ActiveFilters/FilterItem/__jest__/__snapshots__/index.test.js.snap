// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter item of type date 1`] = `
<tr>
  <td>
    Foo
  </td>
  <td>
    Bar
  </td>
  <td
    className="filter-value"
  >
    <span>
      October 21, 1995
    </span>
  </td>
  <td>
    <a
      className="clear-filter"
      onClick={[Function]}
    >
      Clear
    </a>
  </td>
</tr>
`;

exports[`Filter item of type datetime 1`] = `
<tr>
  <td>
    Foo
  </td>
  <td>
    Bar
  </td>
  <td
    className="filter-value"
  >
    <span>
      October 21, 1995, 8:00 PM
    </span>
  </td>
  <td>
    <a
      className="clear-filter"
      onClick={[Function]}
    >
      Clear
    </a>
  </td>
</tr>
`;

exports[`Filter item with child object 1`] = `
<tr>
  <td>
    Foo
  </td>
  <td>
    Bar
  </td>
  <td
    className="filter-value"
  >
    Baz
  </td>
  <td>
    <a
      className="clear-filter"
      onClick={[Function]}
    >
      Clear
    </a>
  </td>
</tr>
`;

exports[`Rendering default filter item 1`] = `
<tr>
  <td />
  <td />
  <td
    className="filter-value"
  />
  <td>
    <a
      className="clear-filter"
      onClick={[Function]}
    >
      Clear
    </a>
  </td>
</tr>
`;
